code/modules/mining/machine_redemption.dm

```diff
@@ -160,11 +174,49 @@
 	else
 		unregister_input_turf() // someone just un-wrenched us, unregister the turf
 
+/obj/machinery/mineral/ore_redemption/can_be_unfasten_wrench(mob/user, silent)
+	if(welded)
+		if(!silent)
+			to_chat(user, span_warning("[src] is welded to the floor!"))
+		return FAILED_UNFASTEN
+
+	return ..()
+	
 /obj/machinery/mineral/ore_redemption/wrench_act(mob/living/user, obj/item/tool)
 	. = ..()
 	default_unfasten_wrench(user, tool)
 	return TOOL_ACT_TOOLTYPE_SUCCESS
+	
+/obj/machinery/mineral/ore_redemption/welder_act(mob/living/user, obj/item/item)
+	..()
+	if(welded)
+		if(!item.tool_start_check(user, amount=0))
+			return TRUE
+		user.visible_message(span_notice("[user.name] starts to cut the [name] free from the floor."), \
+			span_notice("You start to cut [src] free from the floor..."), \
+			span_hear("You hear welding."))
+		if(!item.use_tool(src, user, 20, 1, 50))
```

Comment:
Named variables for 20, 1, and 50. One of those is time, I forget which, turn it to seconds.

Reply:
It's the first one which is delay if I'm understanding the code right (still new to this). I'm not really sure what units it uses but the default delay for unwelding walls is 100. Do you think that would be a suitable delay?

Reply:
he is saying use defines,

20 would be `2 SECONDS`

Reply:
👍